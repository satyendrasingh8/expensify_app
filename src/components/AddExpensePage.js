import React from "react";
import ExpenseForm from './ExpenseForm';
//import ExpenseDashboardPage from "./ExpenseDashboardPage";
import {connect} from 'react-redux';
import {addexpense} from '../actions/expenses';


export class AddExpensePage extends React.Component {
    onSubmit =  (expense) =>{
        // props.dispatch(addexpense(expense));
        this.props.addexpense(expense); 
        this.props.history.push('/');
     } ;
     render() {
return(
    <div>
    <h1>Add Expense </h1>
    <ExpenseForm 
   onSubmit={this.onSubmit}
    />
   </div>

);
   } 

}


    const mapDispatchToProps = (dispatch) => {
    return {
        addexpense:(expense)=> {dispatch(addexpense(expense));   }
      }; 
    };
  
    export default connect(undefined,mapDispatchToProps)(AddExpensePage);