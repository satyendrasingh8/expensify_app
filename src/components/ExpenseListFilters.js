import React from 'react';
import {connect } from 'react-redux';
import {setTextFilter,sortByAmount,sortByDate,setStartDate,setEndtDate} from '../actions/filters';
import {DateRangePicker} from 'react-dates';


class ExpenseListFilters extends React.Component {
state={
    calendarFocused:null
}
  
  onDatesChange = ({startDate,endDate}) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndtDate(endDate));
  }
  onFocusChange =(calendarFocused) => {
this.setState(() => ({calendarFocused}))
  }
    render()
    {
        return (
            <div>
            <input type="text" value={this.props.filters.text} onChange={(e) => {
                this.props.dispatch(setTextFilter(e.target.value));
               
            }} />
            <select
            value={this.props.filters.sortBy}
            onChange={ (e) => {
                if(e.target.value=="date")
                {
             this.props.dispatch(sortByDate())
                }
                else if (e.target.value==="amount")
                {
                   this.props.dispatch(sortByAmount())
                }
            }  }
            
            >
            <option value="date"> Date</option>
            <option value="amount">Amount </option>
            </select>

            <DateRangePicker 
            startDate={this.props.filters.startDate}
            endDate={this.props.filters.endDate}
            onDatesChange={this.onDatesChange}
            focusedInput ={this.state.calendarFocused}
            onFocusChange={this.onFocusChange}
            showClearDates={true}
            numberOfMonths={1}
            isOutsideRange={()=>false}
            />
            </div>
            );
    }
}



const mapStateToProps = (state) => {
    return {
        expenses:state.expenses,
        filters:state.filters
    }
}
export default connect(mapStateToProps)(ExpenseListFilters);