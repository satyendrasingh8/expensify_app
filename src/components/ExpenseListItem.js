import React from 'react';
import {Link} from 'react-router-dom';




 const ExpenseListItem = ({id,description,amount,createdAt}) =>(
    <div>
    <Link to={`/edit/${id}`}>
<h2>{description} </h2>
   </Link>
<p>{amount} - {createdAt} </p>

    </div>
);

export default ExpenseListItem;