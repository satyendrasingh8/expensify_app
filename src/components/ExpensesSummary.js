import React from 'react';
import {connect} from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({expenseCount, expensesTotal}) => {
const formattedExpensesTotal = numeral(expensesTotal/100).format('$0,0.00');
    const expenseWord = expenseCount ===1 ? 'expense' :'expenses';
    return (
  <div>
  <h1> viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal} </h1>
  </div>
    );

};

const mapStateToProps = (state) => {
const visibleExpenses = selectExpenses(state.expenses,state.filters); 
return {
    expenseCount: visibleExpenses.length,
    expensesTotal:selectExpensesTotal(visibleExpenses)  
}
};

export default connect(mapStateToProps)(ExpensesSummary);