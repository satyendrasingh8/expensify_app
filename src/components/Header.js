import React from "react";
import  { NavLink} from 'react-router-dom';

            const Header=() => (
                <header>
                <h1>Expensify </h1>
                <ul> 
               <li> <NavLink to="/" activeClassName="is-active" exact={true}> Dasboard page </NavLink></li>
               <li> <NavLink to="/create" activeClassName="is-active"> create page </NavLink></li>
             
               <li> <NavLink to="/help" activeClassName="is-active"> help page </NavLink></li>
             </ul>
                </header>
            );
                export default Header;