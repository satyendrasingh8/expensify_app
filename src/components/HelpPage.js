import React from "react";
import {Link} from 'react-router-dom';

const HelpPage = () => (
    <div>
     this is from my help page components
     <button><Link to='/'> Go to Dashboard page</Link> </button>

    </div>
    );

    export default HelpPage;