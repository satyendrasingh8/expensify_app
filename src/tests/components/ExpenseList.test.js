import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseList} from '../../components/ExpenseList';
import expenses from  '../fixtures/expenses';


test('should render expenseList from expenses',() => {
   const wrapper = shallow(<ExpenseList expenses={expenses} />);
   expect(wrapper).toMatchSnapshot();
});
