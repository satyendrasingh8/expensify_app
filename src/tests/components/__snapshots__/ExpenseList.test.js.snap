// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenseList from expenses 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpenseList
    expenses={
        Array [
            Object {
              "amount": 1500,
              "createdAt": 0,
              "description": "hello",
              "id": "1",
              "note": "",
            },
            Object {
              "amount": 1300,
              "createdAt": -950400000,
              "description": "hi",
              "id": "2",
              "note": "",
            },
            Object {
              "amount": 400,
              "createdAt": 864000000,
              "description": "hope",
              "id": "3",
              "note": "",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          ExpenseList  
</h1>,
        Array [
          <ExpenseListItem
            amount={1500}
            createdAt={0}
            description="hello"
            id="1"
            note=""
/>,
          <ExpenseListItem
            amount={1300}
            createdAt={-950400000}
            description="hi"
            id="2"
            note=""
/>,
          <ExpenseListItem
            amount={400}
            createdAt={864000000}
            description="hope"
            id="3"
            note=""
/>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ExpenseList  ",
        },
        "ref": null,
        "rendered": "ExpenseList  ",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "amount": 1500,
          "createdAt": 0,
          "description": "hello",
          "id": "1",
          "note": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "amount": 1300,
          "createdAt": -950400000,
          "description": "hi",
          "id": "2",
          "note": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "amount": 400,
          "createdAt": 864000000,
          "description": "hope",
          "id": "3",
          "note": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            ExpenseList  
</h1>,
          Array [
            <ExpenseListItem
              amount={1500}
              createdAt={0}
              description="hello"
              id="1"
              note=""
/>,
            <ExpenseListItem
              amount={1300}
              createdAt={-950400000}
              description="hi"
              id="2"
              note=""
/>,
            <ExpenseListItem
              amount={400}
              createdAt={864000000}
              description="hope"
              id="3"
              note=""
/>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ExpenseList  ",
          },
          "ref": null,
          "rendered": "ExpenseList  ",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "amount": 1500,
            "createdAt": 0,
            "description": "hello",
            "id": "1",
            "note": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "amount": 1300,
            "createdAt": -950400000,
            "description": "hi",
            "id": "2",
            "note": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "amount": 400,
            "createdAt": 864000000,
            "description": "hope",
            "id": "3",
            "note": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should render expenseListItem from expenses 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpenseListItem
    expenses={
        Array [
            Object {
              "amount": 1500,
              "createdAt": 0,
              "description": "hello",
              "id": "1",
              "note": "",
            },
            Object {
              "amount": 1300,
              "createdAt": -950400000,
              "description": "hi",
              "id": "2",
              "note": "",
            },
            Object {
              "amount": 400,
              "createdAt": 864000000,
              "description": "hope",
              "id": "3",
              "note": "",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          to="/edit/undefined"
>
          <h2>
                     
          </h2>
</Unknown>,
        <p>
           - 
           
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <h2>
             
</h2>,
          "to": "/edit/undefined",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            " ",
          ],
          "type": "h2",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            " - ",
            undefined,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          " - ",
          undefined,
          " ",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            to="/edit/undefined"
>
            <h2>
                         
            </h2>
</Unknown>,
          <p>
             - 
             
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <h2>
               
</h2>,
            "to": "/edit/undefined",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              undefined,
              " ",
            ],
            "type": "h2",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              " - ",
              undefined,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            " - ",
            undefined,
            " ",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
